__tests__/app.test.js

```diff
@@ -0,0 +1,150 @@
+        expect(body.article[0]).toEqual({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 100,
+        });
+      });
+  });
+  test("status:404, respond with error message", () => {
+    return request(app)
+      .get("/api/article/500")
+      .expect(404)
+      .then(({ body }) => {
+        console.log(body);
+        const { msg } = body;
+        expect(msg).toBe("file or articole not found");
+      });
+  });
+  test("status:400, responds with an error message when passed a bad user ID", () => {
+    return request(app)
+      .get("/api/articles/notAnID")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+});
+exports.patchParkById = (req, res) => {
+  const { park_id } = req.params;
+  updateParkById(park_id, req.body)
+    .then((park) => {
+      res.status(200).send({ park });
+    })
+    .catch((err) => console.log(err));
+};
+
+describe("GET /api/users", () => {
+  test("200: returns an array of users", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then((res) => {
+        const { body } = res;
+        expect(Array.isArray(body.users)).toBe(true);
+        body.users.forEach((user) => {
+          expect(user).toEqual(
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String),
+            })
+          );
+          expect(Object.keys(user).length).toBe(3);
+        });
+      });
+  });
+});
+
+describe("PATCH /api/articles/:article_id", () => {
+  const articleUpdate = {
+    votes: 3,
+  };
+  test("status:200, responds with the updated article", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send(articleUpdate)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toEqual({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 103,
+        });
+      });
+  });
+  test("status:404, respond with error message", () => {
+    return request(app)
+      .get("/api/article/1000")
+      .expect(404)
+      .then(({ body }) => {
+        console.log(body);
+        const { msg } = body;
+        expect(msg).toBe("file or articole not found");
+      });
+  });
+  test("status:400, responds with an error message when passed a bad user ID", () => {
+    return request(app)
+      .get("/api/articles/notAnID")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+});
```

Comment:
You need a couple more error tests here. What if the articleUpdate object is empty or if the votes is not a number? for example `{votes: 'banana'}`